scheduler/core/statscalculator/statscalculator.py

```diff
@@ -49,44 +49,56 @@ def calculate_timeline_stats(timeline: NightlyTimeline,
         scores_per_program: Dict[ProgramID, float] = {}
         programs = {}
         for night_idx in nights:
+            timeline.time_losses.setdefault(night_idx, {})
             for site in sites:
-                time_losses = {StatCalculator._FAULT_KEY: 0,
-                               StatCalculator._WEATHER_KEY: 0,
-                               StatCalculator._UNSCHEDULE_KEY: 0}
+
+                timeline_time_losses = {StatCalculator._FAULT_KEY: 0,
+                                        StatCalculator._WEATHER_KEY: 0}
+                timeline.time_losses[night_idx].setdefault(site, {})
                 # Gather unsolved interruptions during the night.
                 interruptions = []
                 for entry in timeline.timeline[night_idx][site]:
                     if isinstance(entry.event, InterruptionEvent):
                         interruptions.append(entry.event)
                     elif isinstance(entry.event, InterruptionResolutionEvent):
-                        interruptions.pop()
+                        interruptions.pop()  # remove reported interruption and register the time loss
+                        if isinstance(entry.event, FaultResolutionEvent):
+                            timeline_time_losses[StatCalculator._FAULT_KEY] += int(entry.event.time_loss.total_seconds()/60)
+                        elif isinstance(entry.event, WeatherClosureResolutionEvent):
+                            timeline_time_losses[StatCalculator._WEATHER_KEY] += int(entry.event.time_loss.total_seconds()/60)
```

Comment:
I am not sure if I understand correctly this logic.
If the event is an Interruption it is added to the interruptions array. If the event is an InterruptionResolution, the last interruption event is removed from the array. This means, always the Interrupt and Resolution events will be consecutive or there are cases where an InterruptionResolution is associated to a previous Interruption?

Reply:
They are consecutive. If a fault happens other events might occur ( weather change but not an interruption) but if is weather closure everything is closed so nothing happens in between.